test-site/src/components/MapPin.tsx

```diff
@@ -1,18 +1,55 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
 
-export const MapPin: PinComponent<Location> = () => {
+export interface MapPinProps<T> {
+  coordinate: Coordinate,
+  index: number,
+  mapbox: Map,
+  result: Result<T>
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps<Location>) => {
+  const { coordinate, mapbox, result } = props;
+  const [active, setActive] = useState(false);
+  const popupRef = useRef(new Popup({ offset: 15 })
+    .on('close', () => setActive(false))
+  );
+
+  useEffect(() => {
```

Comment:
should this be a useLayoutEffect since it's doing something visual to the page?

Reply:
useLayoutEffect does not work, and [useEffect is preferred](https://reactjs.org/docs/hooks-reference.html#:~:text=prefer%20the%20standard%20useeffect%20when%20possible%20to%20avoid%20blocking%20visual%20updates)